src/NServiceBus.Core.Tests/Routing/Routers/ReplyConnectorTests.cs

```diff
@@ -12,8 +11,11 @@ public class ReplyConnectorTests
         public async Task Should_set_messageintent_to_reply()
         {
             var router = new ReplyConnector();
+            var replyOptions = new ReplyOptions();
+            replyOptions.SetDestination("Fake");
+
             var context = new TestableOutgoingReplyContext();
-            context.GetOperationProperties().Set(new ReplyConnector.State { ExplicitDestination = "Fake" });
+            context.Extensions.Merge(replyOptions.Context);
```

Comment:
the state is now set using the proper public API (`SetDestination`). It doesn't need to be merged in special ways as the GetOperationProperties extensions has a fallback to lookup the regular context values if it doesn't find an explicit context containing send options.